### ISSUE TITLE
Document revision listing / how to pick correct one for downgrade

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#14680]
It's not terribly uncommon to have to revert (if you don't have a staging environment) when something goes wrong and you approach the end of your maintenance window.

Previously I knew what revision a galaxy was on based on the git log of the migrations folder and seeing what was available in each branch.

Now there's [new documentation for the migrations](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/model/migrations/README.md) [*EDIT: link points to updated readme*] which is fantastic but the revisions were not clear for me. `xyz789` and `abc123` were not clear those should be git-looking hashes, or where I could find those hashes.

1. `upgrade gxy@head`, what does the "head" revision mean, does it mean the latest like it would with git? This might not be clear if you don't know the git git comparison
2. Where can I find the hashes? `sh run_alembic.sh history` apparently, but in this case it works because I want the very first revision, not a more recent one. When on 22.09, how will we know what version was 22.05 at that we should downgrade to? I guess by checking out 22.05 I can find `186d4835587b` but that seems relatively unfriendly.
3. `sh run_alembic.sh current` doesn't list which is galaxy and which is `tsi` (and prints them twice)
   ```
   $ sh run_alembic.sh current
   Activating virtualenv at .venv
   INFO:alembic.runtime.migration:Context impl PostgresqlImpl.
   INFO:alembic.runtime.migration:Will assume transactional DDL.
   d4a650f47a3c (head)
   186d4835587b (head)
   INFO:alembic.runtime.migration:Context impl PostgresqlImpl.
   INFO:alembic.runtime.migration:Will assume transactional DDL.
   d4a650f47a3c (head)
   186d4835587b (head)
   ```
4. What does `base` mean? The term "empty db" here made me worried it'd nuke the database completely, does it do that? Deletes everything? Or is `base` an alias for the revision that I want?
   > Downgrade gxy to base (empty db with empty alembic table):
   > sh run_alembic.sh downgrade gxy@base

xref https://github.com/galaxyproject/galaxy/issues/14710

**Comment by jdavcs** [comment]
I've updated the link to the documentation (it pointed to the old version). And I 100% agree with all the points listed here. `head` is current/most recent; `base` is first (version table is emptied, all alembic revisions reverted) - but it's absolutely not obvious, and the hashes are inconvenient, etc. Working on it.

**Comment by jdavcs** [comment]
I'm working on addressing the main problem (identifying revisions as upgrade/downgrade targets for a given release). Here are comments/updates regarding the other items:

> Previously I knew what revision a galaxy was on based on the git log of the migrations folder and seeing what was available in each branch.

I think that only works if the only changes in the directory are commits with new revisions. However, as we continue to improve our migrations code, we'll be refactoring old revision scripts, adding calls to utility functions, etc. - and git log will show all of those in addition to db schema revision commits. 

> Now there's new documentation ... `xyz789` and `abc123` were not clear those should be git-looking hashes, or where I could find those hashes.

Point taken! Those were stand-ins for alembic hashes (revision identifiers auto-generated by alembic). These examples have been removed; the new documentation uses actual hashes instead.

> 1. `upgrade gxy@head`, what does the "head" revision mean

It's the "latest" revision. The new docs explain it (see [here](https://docs.galaxyproject.org/en/release_22.05/admin/db_migration.html#version), [here](https://docs.galaxyproject.org/en/release_22.05/admin/db_migration.html#dbversion), and [here](https://docs.galaxyproject.org/en/release_22.05/admin/db_migration.html#id2))

> 3. `sh run_alembic.sh current` doesn't list which is galaxy and which is `tsi` (and prints them twice)

I've fixed the printed twice issue: it will do that only when using a separate install database (see #14726)

Unfortunately, I cannot make it to show which revision belongs to which branch (gxy or tsi), at least not easily. That command pulls the revision identifiers from the `alembic_version` table - and that table doesn't store any additional information. It *can* be modified, but I suspect that would require more effort than it's worth (like modifying or augmenting the output of the alembic CLI runner with additional data pulled from the code base or from a modified alembic version table). Also, it will only display 2 revisions when there's no separate install database. And, given that we have only one tsi revision, it should be easy to tell which is which.

> 4. What does `base` mean? The term "empty db" here made me worried it'd nuke the database completely, does it do that? Deletes everything? Or is `base` an alias for the revision that I want?

Yeah, not obvious at all, sorry about that! The new docs explain it [here](https://docs.galaxyproject.org/en/release_22.05/admin/db_migration.html#id3): "Specifying base as the revision identifier will downgrade both branches, gxy and tsi, to their initial state prior to any revisions; the alembic_version table will be empty."

**Comment by jdavcs** [comment]
OK, how's this for a solution:
`sh manage_db.sh upgrade 22.05` or `sh manage_db.sh upgrade release_22.05` to move up to 22.05
and
`sh manage_db.sh downgrade 22.01` or `sh manage_db.sh downgrade release_22.01` to move back to 22.01

(Credit goes to @mvdbeek for proposing the simple syntax.)

For each new release, we simply add 2 tags (e.g. `release_23.1` and `23.1`) that we can use instead of the actual revision hash.

Note: proper downgrading to 22.01 is not enforced: you can simply check out the 22.01 branch, and it will work IFF:
a) the database has the `migrate_version` table which contains the value 180; and
b) the database schema has not been modified past version 180: this is what we are not verifying (yet), and this is what running `sh manage_db.sh downgrade 22.01` ensures.

Implemented in #14726. If this is acceptable, I'll update the docs accordingly.

**Comment by jdavcs** [comment]
Fixed in #14730
